gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/auth/GcsDelegationTokens.java

```diff
@@ -58,6 +62,11 @@
   /** Active Delegation token. */
   private Token<DelegationTokenIdentifier> boundDT;
 
+  private DelegationTokenInstantiationStrategy tokenInstantiationStrategy =
+      DelegationTokenInstantiationStrategy.INSTANCE_PER_SERVICE;
+  private static ConcurrentHashMap<Optional<String>, Token<DelegationTokenIdentifier>>
```

Comment:
Also do we need to make it static, so it's shared between instances of these class even though they can be used by different FS objects?

Reply:
Yes, that's the intention of this cache - to be able to share tokens across multiple GCS filsystems.